VERSION-RANGE-SPEC.rst

```diff
@@ -0,0 +1,989 @@
+
+By convention the versioning scheme **should** be the same as the ``Package URL``
+package type for a given package ecosystem. It is OK to have other schemes
+beyond the purl type. A scheme could be specific to a single package name.
+
+The ``<versioning-scheme>`` is followed by a slash "/".
+
+
+``<version-constraint>``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+After the ``<versioning-scheme>`` and "/" there are one or more
+``<version-constraint>`` separated by a pipe "|". The pipe "|" has no special
+meaning beside being a separator.
+
+Each  ``<version-constraint>`` of this list is either a single ``<version>`` as
+in ``1.2.3`` for example or the combination of a ``<comparator>`` and a ``<version>`` as in
+``>=2.0.0`` using this syntax::
+
+    <comparator><version>
+
+A single version that means that a version equal to this version satisfies the
+range spec. Equality is based on the equality of two normalized version strings
+according to their versioning scheme. For most schemes, this is a simple string
+equality. But schemes can specify normalization and rules for equality such as
+``pypi`` with PEP440. 
+
+
+The special star "*" comparator matches any version. It must be used **alone**
+exclusive of any other constraint and must not be followed by a version. For
+example "vers:deb/\*" represent all the versions of a Debian package. This
+includes past, current and possible future versions.
+
+
+Otherwise, the ``<comparator>`` is one of these comparison operators:
+
+- "!=": Version exclusion or inequality comparator. This means a version must
+  not be equal to the provided version that must be excluded from the range.
+  For example: "!=1.2.3" means that version "1.2.3" is excluded.
+
+- "<", "<=": Lesser than or lesser-or-equal version comparators point to all
+  versions less than or equal to the provided version.
+  For example "<=1.2.3" means less than or equal to "1.2.3".
+
+- ">", ">=": Greater than or greater-or-equal version comparators point to
+  all versions greater than or equal to the provided version.
+  For example ">=1.2.3" means greater than or equal to "1.2.3".
+
+
+The ``<versioning-scheme>`` defines:
+
+- how to compare two version strings using these comparators, and
+
+- the structure of a version string such as "1.2.3" if any. For instance, the
+  ``semver`` specification for version numbers  defines a version as composed
+  primarily of three dot-separated numeric segments named major, minor and patch.
+
+
+
+Normalized, canonical representation and validation
+-----------------------------------------------------
+
+The construction and validation rules are designed such that a ``vers`` is
+easier to read and understand by human and straight forward to process by tools,
+attempting to avoid the creation of empty or impossible version ranges.
+
+- Spaces are not significant and removed in a canonical form. For example
+  "<1.2.3|>=2.0" and " <  1.2. 3 | > = 2  . 0" are equivalent.
```

Comment:
The pipe should be seen as a union here. Think of these as tags or signposts that define slices or intervals in the ordered list of all the versions of a package. 

Maven `(,1.0],[1.2,)` would come out as `vers:maven/<=1.0|>=1.2` so it can be easily represented.